.circleci/config.yml

```
@@ -471,40 +487,30 @@ jobs:
       CC: clang
       CXX: clang++
       MAKEFLAGS: -j 10
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
+    <<: *steps_b_ubu
 
-  b_ubu_asan_clang: &b_ubu_asan_clang
-    docker:
-      - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
-    environment:
-      CC: clang
-      CXX: clang++
-      CMAKE_OPTIONS: -DSANITIZE=address
-      MAKEFLAGS: -j 3
-    steps:
-      - checkout
-      - run: *run_build
-      - store_artifacts: *artifacts_solc
-      - persist_to_workspace: *artifacts_executables
-
-  b_ubu_ubsan_clang: &b_ubu_ubsan_clang
+  b_ubu_au_clang:
+    parameters:
+      cmake_option:
+        type: string
+      gitter_notify:
+        type: boolean
+        default: false
     docker:
       - image: << pipeline.parameters.ubuntu-2004-clang-docker-image >>
     environment:
       CC: clang
       CXX: clang++
-      CMAKE_OPTIONS: -DSANITIZE=undefined
+      CMAKE_OPTIONS: << parameters.cmake_option >>
```

I'd make the parameter more specific.
```suggestion
      CMAKE_OPTIONS: -DSANITIZE=<< parameters.sanitizer >>
```
 `cmake_options` as a parameter would make sense too if this job was more generic but for now it's meant only for sanitizers.

---

Unless you decide to go for https://github.com/ethereum/solidity/pull/11896#discussion_r711180445. Then leaving it as is would make sense.

---

I left the more generic version for now.